soci/ztoc.go

```diff
@@ -140,6 +142,8 @@ func ExtractFile(r *io.SectionReader, config *FileExtractConfig) ([]byte, error)
 			if j == 0 && firstSpanHasBits {
 				rangeStart--
 			}
+			mu.Lock()
```

Comment:
I'm a little confused by the need for this mutex. Theoretically, writes to the slice ranges should not overlap.

Reply:
When `firstSpanHasBits` holds true for the certain range, its `rangeStart`, which was initially `starts[<span_id>]` gets decremented (https://github.com/awslabs/soci-snapshotter/blob/main/soci/ztoc.go#L141). Since every `ReadAt` operation goes in its own goroutine, there is a likelihood that `rangeStart` with `firstSpanHasBits` will also be intersecting with preceding interval.

Reply:
But we only decrement it for the first span. No other spans' `rangeStart` gets decremented.

Reply:
This can happen if you are extracting multiple files from the same `tar.gz` as described in TestDecompress. In this case there can be the situation like this, where the span ranges for the current and the following file intersect: 
```
[vkuzniet] spanId=0, rangeStart=3940936, rangeEnd=3953290
[vkuzniet] spanId=1, rangeStart=3953290, rangeEnd=3965604
[vkuzniet] spanId=2, rangeStart=3965604, rangeEnd=3977923
[vkuzniet] spanId=6, rangeStart=4014846, rangeEnd=4021196
[vkuzniet] spanId=4, rangeStart=3990223, rangeEnd=4002533
[vkuzniet] spanId=3, rangeStart=3977923, rangeEnd=3990223
[vkuzniet] spanId=5, rangeStart=4002533, rangeEnd=4014846
[vkuzniet] spanId=0, rangeStart=4014845, rangeEnd=4021196
--- FAIL: TestDecompress (8.39s) testing.go:1319: race detected during execution of test
```